third_party/move/move-model/src/model.rs

```diff
@@ -2671,6 +2677,13 @@ impl<'env> ModuleEnv<'env> {
         self.env.file_id_is_primary_target.contains(&file_id)
     }
 
+    /// Returns true if both modules are defined in the same package.
+    /// Requires: one of the modules is a primary target.
+    pub fn in_same_package(&self, other: &'env Self) -> bool {
+        debug_assert!(self.is_primary_target() || other.is_primary_target());
+        self.is_primary_target() == other.is_primary_target()
```

Comment:
This logic is still broken, if they are not both primary target, they are not in the same package.

You probably want to remove this function and do the test directly at caller side. The test should work as follows:

- You only check visibility for is_primary_target modules in the first place.
- Then the call target for package visibility is fine if the target module is_primary_target as well

Reply:
> If they are not both primary target, they are not in the same package.

Yes. So the method should not be called in such cases, which is documented and asserted.